Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36307&attempt_id=95a8ca22-4484-410a-b960-9abd439a71ef)

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/9avjr40zgds5rycm2ndw9imrfpkb3vga-CF-osx-10.10.5-private/Library/Frameworks/CoreFoundation.framework/Headers/CFUserNotification.h:123:78: error: use of old-style cast [-Werror,-Wold-style-cast]
CF_INLINE CFOptionFlags CFUserNotificationPopUpSelection(CFIndex n) {return ((CFOptionFlags)(n << 24));}
                                                                             ^              ~~~~~~~~~
11 errors generated.
make[2]: *** [src/CMakeFiles/keepassx_core.dir/build.make:519: src/CMakeFiles/keepassx_core.dir/core/ScreenLockListenerPrivate.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:197: src/CMakeFiles/keepassx_core.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
builder for '/nix/store/bdzif384690l72s636hp08pkix2a96db-keepassxc-2.3.0.drv' failed with exit code 2
error: build of '/nix/store/bdzif384690l72s636hp08pkix2a96db-keepassxc-2.3.0.drv' failed
```
</p></details>

--- a/pkgs/applications/misc/keepassx/community.nix
+++ b/pkgs/applications/misc/keepassx/community.nix
@@ -34,6 +34,8 @@ stdenv.mkDerivation rec {
     sha256 = "1zch1qbqgphhp2p2kvjlah8s337162m69yf4y00kcnfb3539ii5f";
   };
 
+  NIX_CFLAGS_COMPILE = stdenv.lib.optionalString stdenv.cc.isClang "-Wno-old-style-cast";
+
   patches = [ ./darwin.patch ];
 
   cmakeFlags = [

